File: tests/fixtures/scenario_xlang_php_sqli/vulnerable.php
<?php
// Vulnerable: classic PHP SQL injection via $_GET concatenated into mysqli_query.
// CWE-89 / OWASP A03:2021 — Injection.

$mysqli = new mysqli("localhost", "user", "pass", "app");

$user_id = $_GET['id'];
$query = "SELECT id, name, email FROM users WHERE id = " . $user_id;
$result = $mysqli->query($query);

while ($row = $result->fetch_assoc()) {
    echo htmlspecialchars($row['name']) . "\n";
}

File: tests/fixtures/scenario_04_sqli_diff/vulnerable.py
# Intentionally vulnerable fixture: classic string-concatenation SQL injection.
# Semgrep's auto config should flag this; the LLM exploitability step should
# rate it high.

import sqlite3
from flask import Flask, request

app = Flask(__name__)
DB = sqlite3.connect(":memory:")


@app.route("/user")
def get_user():
    user_id = request.args.get("id")
    # Direct concatenation of user input into SQL is the canonical
    # CWE-89 / OWASP A03 case.
    query = "SELECT * FROM users WHERE id = " + user_id
    cur = DB.cursor()
    cur.execute(query)
    return {"row": cur.fetchone()}
